crates/resource-resolver/src/lib.rs

```diff
@@ -0,0 +1,192 @@
+///
+/// Linux allows for the creation of symlinks to any file. Accessing the
+/// symlink will fail if the user who owns the symlink does not have the
+/// proper permissions on the original file.
+///
+/// Linux additionally provides a kernel hardening feature since version
+/// 3.6 (30 September 2012). Most distributions since then have enabled
+/// the protection (setting `fs.protected_hardlinks = 1`) by default, which
+/// means that a vast majority of desktop Linux users should have it enabled.
+/// **The feature prevents the creation of hardlinks that the user does not own
+/// or have read/write access to.** [See the patch that enabled this].
+///
+/// [Hard Link]: https://en.wikipedia.org/wiki/Hard_link
+/// [See the patch that enabled this]: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=800179c9b8a1e796e441674776d11cd4c05d61d7
+pub fn current_exe() -> Result<PathBuf> {
+    starting_binary::STARTING_BINARY
+        .cloned()
+        .map_err(|e| Error::Io("Can't detect the path of the current exe".to_string(), e))
+}
+
+/// Retreive the resource path of your app, packaged with cargo packager.
+/// This function behave the same as [`resource_dir`], except it accepts
+/// a parameter that will be happened to the resource path when no packaging format
+/// is used.
+///
+/// Example: You want to include the folder `crate/resource/icons/`.
+///
+/// - With `cargo run` command, you will have to execute
+///     `resource_dir().unwrap().join("resource/icons/")` to get the path.
+/// - With any other formats, it will be `resource_dir().unwrap().join("icons/")`.
+///
+/// ```
+/// use cargo_packager_resource_resolver as resource_resolver;
+/// use resource_resolver::{PackageFormat, resource_dir_with_suffix};
+///
+/// resource_dir_with_suffix(PackageFormat::None, "resource").unwrap().join("icons/");
+/// ```
+pub fn resource_dir_with_suffix(package_format: PackageFormat, suffix: &str) -> Result<PathBuf> {
+    match package_format {
+        PackageFormat::None => {
+            let root_crate_dir = env::var("CARGO_MANIFEST_DIR")
+                .map_err(|e| {
+                    match e {
+                        env::VarError::NotPresent => {
+                            Error::Env("PackageFormat::None was use, but CARGO_MANIFEST_DIR environnement variable was not defined".to_string())
+                        },
+                        _ => Error::Var("Can't access CARGO_MANIFEST_DIR environnement variable".to_string(), e)
+                    }
+                })?;
+            Ok(PathBuf::from(root_crate_dir).join(suffix))
+        }
+        PackageFormat::App | PackageFormat::Dmg => {
+            let exe = current_exe()?;
+            let exe_dir = exe.parent().unwrap();
+            exe_dir
+                .join("../Resources")
+                .canonicalize()
+                .map_err(|e| Error::Io("".to_string(), e))
+        }
+        PackageFormat::Wix => Err(Error::UnsupportedPlatform),
```

Comment:
This should be the same as `Nsis` format.

Reply:
Alright, what about AppImage?